testsuite/MDAnalysisTests/coordinates/test_dcd.py

```diff
@@ -65,420 +87,233 @@ def test_with_statement(self):
-        self.dcd[15]  # index is 0-based and frames are 0-based
-        assert_equal(self.universe.trajectory.frame, 15, "wrong frame number")
-
-    def test_time(self):
-        self.dcd[15]  # index is 0-based and frames are 0-based
-        assert_almost_equal(self.universe.trajectory.time,
-                            15.0,
-                            5,
-                            err_msg="wrong time of frame")
-
-    def test_volume(self):
-        assert_almost_equal(self.ts.volume, 0.0, 3,
-                            err_msg="wrong volume for unitcell (no unitcell "
-                            "in DCD so this should be 0)")
-
-    def test_timeseries_slicing(self):
-        # check that slicing behaves correctly
-        # should  before issue #914 resolved
-        x = [(0, 1, 1), (1,1,1), (1, 2, 1), (1, 2, 2), (1, 4, 2), (1, 4, 4),
-             (0, 5, 5), (3, 5, 1), (None, None, None)]
-        for start, stop, step in x:
-            yield self._slice_generation_test, start, stop, step
-
-    def test_backwards_stepping(self):
-        x = [(4, 0, -1), (5, 0, -2), (5, 0, -4)]
-        for start, stop, step in x:
-            yield self._failed_slices_test, start, stop, step
-
-    def _slice_generation_test(self, start, stop, step):
-        self.u = mda.Universe(PSF, DCD)
-        ts = self.u.trajectory.timeseries(self.u.atoms)
-        ts_skip = self.u.trajectory.timeseries(self.u.atoms, start, stop, step)
-        assert_array_almost_equal(ts[:, start:stop:step,:], ts_skip, 5)
-
-    @knownfailure
-    def _failed_slices_test(self, start, stop, step):
-        self.u = mda.Universe(PSF, DCD)
-        ts = self.u.trajectory.timeseries(self.u.atoms)
-        ts_skip = self.u.trajectory.timeseries(self.u.atoms, start, stop, step)
-        assert_array_almost_equal(ts[:, start:stop:step,:], ts_skip, 5)
-
-
-def test_DCDReader_set_dt(dt=100., frame=3):
+class TestDCDWriter(BaseWriterTest):
+    def __init__(self, reference=None):
+        if reference is None:
+            reference = DCDReference()
+        super(TestDCDWriter, self).__init__(reference)
+
+
+def test_write_random_unitcell(tmpdir):
+    with tmpdir.as_cwd():
+        testname = 'test.dcd'
+        rstate = np.random.RandomState(1178083)
+        random_unitcells = rstate.uniform(
+            high=80, size=(98, 6)).astype(np.float64)
+
+        u = mda.Universe(PSF, DCD)
+        with mda.Writer(testname, n_atoms=u.atoms.n_atoms) as w:
+            for index, ts in enumerate(u.trajectory):
+                u.atoms.dimensions = random_unitcells[index]
+                w.write(u.atoms)
+
+        u2 = mda.Universe(PSF, testname)
+        for index, ts in enumerate(u2.trajectory):
+            assert_array_almost_equal(ts.dimensions, random_unitcells[index],
+                                      decimal=5)
+
+
+################
+# Legacy tests #
+################
+
+@pytest.fixture(scope='module')
+def universe_dcd():
+    return mda.Universe(PSF, DCD)
```

Comment:
Looks to work fine now.  This has some limitations of course but here I only ever iterate over a whole trajectory. For that it should work fine because we reset the frame correctly in the basereader code.